File: FitnessWebApi/FitnessWebApi/_Services/UserService.cs
namespace FitnessWebApi._Services
{
	public interface IUserService
	{
		public Task<List<StaticUserResponse>> GetAll();
		public Task<DirectUserResponse> GetById(int id);
		public Task<DirectUserResponse> Create(UserRequest request);
		public Task<DirectUserResponse> Update(int id, UserRequest request);
	}

	public class UserService : IUserService
	{
		private readonly IUserRepository _repository;
		private readonly IMapper m_mapper;

		public UserService(IUserRepository repository, IMapper mapper)
		{
			_repository = repository;
			m_mapper = mapper;
		}

		public async Task<List<StaticUserResponse>> GetAll()
		{
			List<User> users = await _repository.GetAll();
			if(users != null)
			{
				return users.Select(user => m_mapper.Map<User, StaticUserResponse>(user)).ToList();
			}

			return null;
		}

		public async Task<DirectUserResponse> GetById(int id)
		{
			User user = await _repository.GetById(id);
			if (user != null)
			{
				return m_mapper.Map<DirectUserResponse>(user);
			}

			return null;
		}

		public async Task<DirectUserResponse> Create(UserRequest request)
		{
			User user = await _repository.Create(m_mapper.Map<User>(request));
			if (user != null)
			{
				return m_mapper.Map<DirectUserResponse>(user);
			}

			return null;
		}

		public async Task<DirectUserResponse> Update(int id, UserRequest request)
		{
			User user = await _repository.Update(id, m_mapper.Map<User>(request));
			if (user != null)
			{
				return m_mapper.Map<DirectUserResponse>(user);
			}

			return null;
		}
	}
}

File: Nicklas-Fitness-App/app/src/main/java/com/example/nicklastest/services/UserService.java
package com.example.nicklastest.services;

import com.example.nicklastest.models.User.DirectUserResponse;
import com.example.nicklastest.models.User.StaticUserResponse;
import com.example.nicklastest.models.User.UserRequest;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface UserService {
    @GET()
    Observable<List<StaticUserResponse>> GetAll();

    @GET("{id}")
    Observable<DirectUserResponse> GetById(@Path("id") int id);

    @POST()
    Observable<DirectUserResponse> Create(@Body UserRequest request);

    @PUT("{id}")
    Observable<DirectUserResponse> Update(@Path("id") int id, @Body UserRequest request);
}
